setup/vars.tf

```diff
@@ -29,6 +40,20 @@ variable "project_id" {
   }
 }
 
+# This variable is used to override default naming behavior.  By default, names
+# are scoped by terraform.workspace if not "default", otherwise it will be set
+# to "prod" if is_production is true or "nonprod" if false
+variable "environment" {
+  type        = string
+  default     = "default"
+  description = "The name of the environment"
+
+  validation {
+    condition     = can(regex("^[[:alpha:]][[:alnum:]]{0,10}$", var.environment))
+    error_message = "environment can only contain letters and numbers, must start with a letter, and must be 10 or fewer characters"
+  }
+}
+
 variable "is_production" {
   type        = bool
   default     = false
```

Comment:
with this default value, will environment ever come into play?

Reply:
As above, `is_production` is independent from `environment`.  It's perfectly possible to have "production" environments named "prod", "production", or even "<my-ldap>".

There aren't many cases where the `environment` var will be required, but I don't like the idea of having default values that can't be overridden somehow.  It's not unreasonable that someone might want to deploy from their own personal workspace but need to pick an environment name like "prodtest" due to permissions or org polices.

Reply:
resolving bc not relevant